pallets/price-oracle/src/types.rs

```diff
@@ -0,0 +1,222 @@
+// This file is part of Basilisk-node.
+
+// Copyright (C) 2020-2021  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use codec::{Decode, Encode};
+use frame_support::sp_runtime::traits::{CheckedAdd, CheckedDiv, CheckedMul, Zero};
+use frame_support::sp_runtime::RuntimeDebug;
+pub use primitives::{Balance, Price};
+use sp_std::iter::Sum;
+use sp_std::ops::{Add, Index, IndexMut};
+use sp_std::prelude::*;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+/// A type representing data produced by a trade.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(RuntimeDebug, Encode, Decode, Copy, Clone, PartialEq, Eq, Default)]
+pub struct PriceEntry {
+	pub price: Price,
+	pub trade_amount: Balance,
+	pub liquidity_amount: Balance,
+}
+
+impl Add for PriceEntry {
+	type Output = Self;
+	fn add(self, other: Self) -> Self {
+		Self {
+			price: self.price.add(other.price),
+			trade_amount: self.trade_amount.add(other.trade_amount),
+			liquidity_amount: self.liquidity_amount.add(other.liquidity_amount),
+		}
+	}
+}
+
+impl Zero for PriceEntry {
+	fn zero() -> Self {
+		Self {
+			price: Price::zero(),
+			trade_amount: Balance::zero(),
+			liquidity_amount: Balance::zero(),
+		}
+	}
+
+	fn is_zero(&self) -> bool {
+		self == &PriceEntry::zero()
+	}
+}
+
+impl Add for &PriceEntry {
+	type Output = PriceEntry;
+	fn add(self, other: Self) -> Self::Output {
+		PriceEntry {
+			price: self.price.add(other.price),
+			trade_amount: self.trade_amount.add(other.trade_amount),
+			liquidity_amount: self.liquidity_amount.add(other.liquidity_amount),
+		}
+	}
+}
+
+impl<'a> Sum<&'a Self> for PriceEntry {
+	fn sum<I>(iter: I) -> Self
+	where
+		I: Iterator<Item = &'a Self>,
+	{
+		iter.fold(
+			PriceEntry {
+				price: Price::zero(),
+				trade_amount: Balance::zero(),
+				liquidity_amount: Balance::zero(),
+			},
```

Comment:
```suggestion
			PriceEntry::zero(),
```

or if default was implemented: 

```suggestion
			PriceEntry::default(),
```